### PR TITLE
pull GradebookHeader out of monolith

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Pull GradebookHeader component out of monolith and clean up some props.

JIRA: [JIRA-5662](https://openedx.atlassian.net/browse/JIRA-5662)

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [x] Bumped version number [package.json](../package.json)

**Testing Instructions**

Basic smoke test should suffice.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by nsprenkle — APPROVED** [review]
Looks good, but a UX note: when the page comes up it starts with the default "you don't have permission to view this page" and halts a second before firing of requests which _actually_ check this. I don't remember it doing this in the past. It's not a blocker, but gives me a bit of pause each time I think I've broken the page for a second or so.
